goal_src/jak3/engine/collide/collide-cache.gc

```diff
@@ -7,3 +7,898 @@
+            )
+          )
+        )
+      (.ftoi.vf vf3 vf1)
+      (nop!)
+      (.ftoi.vf vf4 vf2)
+      (set! (-> this num-tris) 0)
+      (nop!)
+      (.svf (&-> this collide-box min quad) vf1)
+      (nop!)
+      (.svf (&-> this collide-box max quad) vf2)
+      (nop!)
+      (.svf (&-> this collide-box4w min quad) vf3)
+      (nop!)
+      (.svf (&-> this collide-box4w max quad) vf4)
+      (nop!)
+      (.svf (&-> arg0 bbox4w min quad) vf3)
+      (nop!)
+      (.svf (&-> arg0 bbox4w max quad) vf4)
+      (nop!)
+      (set! (-> this num-prims) 0)
+      (nop!)
+      (set! (-> this collide-with) (the-as collide-spec v1-4))
+      )
+    0
+    (when (logtest? (-> arg0 collide-with) (collide-spec backgnd))
+      (let* ((a0-3 this)
+             (t9-0 (method-of-object a0-3 fill-from-bg))
+             (a1-2 (method-of-type collide-hash drawable-method-11))
+             (a2-1 collide-list-fill-bg-using-box)
+             (a3-0 arg0)
+             )
+        (t9-0 a0-3 (the-as (function collide-hash int collide-list collide-query int) a1-2) a2-1 a3-0)
+        (dotimes (s4-0 (-> *level* length))
+          (let ((v1-15 (-> *level* level s4-0)))
+            (when (= (-> v1-15 status) 'active)
+              (let ((a0-6 (-> v1-15 bsp hfrag-drawable)))
+                (if (nonzero? a0-6)
+                    ((-> (the-as process-drawable (-> a0-6 type)) event-hook)
+                     (the-as process a0-6)
+                     (the-as int this)
+                     (the-as symbol arg0)
+                     (the-as event-message-block a3-0)
+                     )
+                    )
+                )
+              )
+            )
+          )
+        )
+      (let ((a0-7 (-> this num-tris)))
+        (when (> a0-7 0)
+          (let ((v1-22 (-> this prims))
+                (a1-9 *collide-shape-prim-backgnd*)
+                )
+            (set! (-> v1-22 0 num-tris) (the-as uint a0-7))
+            (set! (-> v1-22 0 prim) a1-9)
+            (set! (-> this num-prims) 1)
+            (set! (-> v1-22 0 first-tri) (the-as uint 0))
+            (set! (-> v1-22 0 ccache) this)
+            (set! (-> v1-22 0 prim-core world-sphere quad) (-> a1-9 prim-core world-sphere quad))
+            (set! (-> v1-22 0 prim-core quad 1) (-> a1-9 prim-core quad 1))
+            )
+          )
+        )
+      (+! (-> *collide-stats* output) (-> this num-tris))
+      )
+    (when (logtest? (-> arg0 collide-with) (collide-spec water))
+      (let ((v1-29 (-> arg0 ignore-process0)))
+        (if v1-29
+            (fill-from-water this (-> (the-as process-drawable v1-29) water))
+            )
+        )
+      )
+    (if (logtest? (-> arg0 collide-with) (collide-spec hit-by-player-list hit-by-others-list player-list))
+        (fill-from-fg-boxes this)
+        )
+    0
+    (none)
+    )
+  )
+
+(defmethod fill-using-line-sphere ((this collide-cache) (arg0 collide-query))
+  (local-vars (v1-11 float) (v1-20 float))
```

Comment:
I think `v1-11` needs to be an `int` or `uint` here - the decompiler gets this wrong sometimes.